### ISSUE TITLE
Venice Unleashed

### ISSUE THREAD

**Issue by D3thw0lf** [parkervcp/eggs#873]
Service: Battlefield 3

Does this expand an already existing service: N

Link to game: https://www.ea.com/de-de/games/battlefield/battlefield-3

Links for server downloads: https://veniceunleashed.net/

Links for install steps/docs: https://docs.veniceunleashed.net/hosting/setup-linux/

Additional Stuff:
_This is just Info I've seen from their FAQ!_

> **Is it legal?**
> Yes. VU does not modify nor distribute any EA/DICE assets and functions separately from the vanilla BF3 client. VU is fully legal under EU interoperability laws.

---
Would love to hear back and ofc. if someone is way more experienced to create the docker/egg (if possible) etc. ... would love to see that added. <3

**Comment by gOOvER** [comment]
You can't not really Download BF3 and upload via scp is not really a way.

**Comment by itsmemac** [comment]
Would love to have this egg

@gOOvER Just have the basic files from the repo downloaded and we supply the game file (30gb+) ?

Like FiveM egg?

Heard that EA hates docker

**Comment by gOOvER** [comment]
I will not such a Egg

**Comment by GhostingGodzilla** [comment]
I will love to have this

**Comment by gOOvER** [comment]
> I will love to have this

Go on and create an egg :)

**Comment by GhostingGodzilla** [comment]
I have no idea even how to do this

**Comment by parkervcp** [comment]
I won't support it. They never released a public server for battlefield 3 and I don't want to be the guy who gets sued for making it easy to host.

**Comment by Nooobiecake** [comment]
> I won't support it. They never released a public server for battlefield 3 and I don't want to be the guy who gets sued for making it easy to host.

Well its not like you are hosting a bf3 server you are hosting a Venice unleashed server :)

I have had no time to look into creating eggs yet. but ill try it later when i have some time..

**Comment by xardoniak** [comment]
Keen for an egg!

**Comment by parkervcp** [comment]
I will consider adding this.

Looking at this and the hardest part will be activating the game. Also shipping the games files around is illegal. So you will need to figure that out on your own. It will also require users putting an email and password in as a variable to activate the server.

I don't see a host being able to use this in any way that is viable. This will basically be a "self host only" kind of egg.

**Comment by xardoniak** [comment]
Looks like someone has already created a docker image for this, if it helps at all https://community.veniceunleashed.net/t/vu-pterodacyl-egg/1931

**Comment by parkervcp** [comment]
I already have a wine image. I am also pretty sure I already have this working I am just in the testing phase.

**Comment by xardoniak** [comment]
@parkervcp more than happy to spin up a test server if that helps at all

**Comment by parkervcp** [comment]
Yeah this takes a lot to actually get working and needing to manually upload a copy of your server is a tedious process.

**Comment by parkervcp** [comment]
Some notes on setup.

I tried using my generic wine image I just pushed. #1125 

 I make no promises as I never got the game files uploaded.

custom startup is below. Needs variables added for username and password potentially.

```
if [ ! -f /home/container/activated ]; then; wine ~/vu/client/vu.com -gamepath ~/bf3 -activate -o_mail ${EMAIL} -o_pass ${PASSWORD}; touch /home/container/activated; else; wine ~/vu/client/vu.com -gamepath ~/bf3 -serverInstancePath "$(winepath -w ~/vu/instance)" -server -dedicated; fi
```

**Comment by empmdk** [comment]
> Some notes on setup.
> 
> I tried using my generic wine image I just pushed. #1125
> 
> I make no promises as I never got the game files uploaded.
> 
> custom startup is below. Needs variables added for username and password potentially.
> 
> ```
> if [ ! -f /home/container/activated ]; then; wine ~/vu/client/vu.com -gamepath ~/bf3 -activate -o_mail ${EMAIL} -o_pass ${PASSWORD}; touch /home/container/activated; else; wine ~/vu/client/vu.com -gamepath ~/bf3 -serverInstancePath "$(winepath -w ~/vu/instance)" -server -dedicated; fi
> ```

I've uploaded all the files to my server and when I run with this startup command  I get a syntax error and it crashes.
`/entrypoint.sh: eval: line 85: syntax error near unexpected token `;'`

**Comment by itsmemac** [comment]
> > Some notes on setup.
> > I tried using my generic wine image I just pushed. #1125
> > I make no promises as I never got the game files uploaded.
> > custom startup is below. Needs variables added for username and password potentially.
> > ```
> > if [ ! -f /home/container/activated ]; then; wine ~/vu/client/vu.com -gamepath ~/bf3 -activate -o_mail ${EMAIL} -o_pass ${PASSWORD}; touch /home/container/activated; else; wine ~/vu/client/vu.com -gamepath ~/bf3 -serverInstancePath "$(winepath -w ~/vu/instance)" -server -dedicated; fi
> > ```
> 
> I've uploaded all the files to my server and when I run with this startup command I get a syntax error and it crashes.
> 
> /entrypoint.sh: eval: line 85: syntax error near unexpected token `;'

Working startup below
`if [ ! -f /home/container/activated ]; then wine ~/vu/client/vu.com -gamepath ~/bf3 -activate -o_mail ${EMAIL} -o_pass ${PASSWORD}; touch /home/container/activated; else wine ~/vu/client/vu.com -gamepath ~/bf3 -serverInstancePath "$(winepath -w ~/vu/instance)" -server -dedicated; fi`

**Comment by itsmemac** [comment]
Only thing this needs is a dockerimage with vcredist installed using wine for the egg to work with venice unleashed
Other than that, all factors are covered but needs a bit tweaking of wine generic egg with variables for email and password
and install script modified to make the required directories

**Comment by itsmemac** [comment]
> 

The current implementation which i have with the egg is 2 types where in 1, user can have a central directory and then use ptero's mount feature to use the same files for as many servers as you want (i use the mount version mainly and run 4 servers hence saving the files) 

Mount based egg is for personal use where users manage the server completely

Without mount egg will be useful for hosts where they just give the vu installed server to the user and when the user get the server, they will have to fill in their email and password in startup tab, supply gamefiles to `bf3` directory and provide server licence from vu website. Thus making the egg safer to host since Hosts are not gonna supply the files and doesn't affect any TOS of EA.

**Comment by Exophobias** [comment]
> > > I will consider adding this.
> > > Looking at this and the hardest part will be activating the game. Also shipping the games files around is illegal. So you will need to figure that out on your own. It will also require users putting an email and password in as a variable to activate the server.
> > > I don't see a host being able to use this in any way that is viable. This will basically be a "self host only" kind of egg.
> > 
> > 
> > I would like to point you that I have a 64GB server with a Ryzen 5 that I use for personal server hosting purposes. I have Pterodactyl installed for easy game server management. I'm not a host, but I would like an easy way to get Venice Unleashed base server files. Then as the instance owner, I can upload the game server files and put in my activation and pass to validate server.
> 
> The current implementation which i have with the egg is 2 types where in 1, user can have a central directory and then use ptero's mount feature to use the same files for as many servers as you want (i use the mount version mainly and run 4 servers hence saving on disk usage)
> 
> Mount based egg is for personal use where users manage the server completely
> 
> Without mount egg will be useful for hosts where they just give the vu installed server to the user and when the user get the server, they will have to fill in their email and password in startup tab, supply gamefiles to `bf3` directory and provide server licence from vu website. Thus making the egg safer to host since Hosts are not gonna supply the files and doesn't affect any TOS of EA.

Would you provide me with your egg(s)?

**Comment by itsmemac** [comment]
> > > > I will consider adding this.
> > > > Looking at this and the hardest part will be activating the game. Also shipping the games files around is illegal. So you will need to figure that out on your own. It will also require users putting an email and password in as a variable to activate the server.
> > > > I don't see a host being able to use this in any way that is viable. This will basically be a "self host only" kind of egg.
> > > 
> > > 
> > > I would like to point you that I have a 64GB server with a Ryzen 5 that I use for personal server hosting purposes. I have Pterodactyl installed for easy game server management. I'm not a host, but I would like an easy way to get Venice Unleashed base server files. Then as the instance owner, I can upload the game server files and put in my activation and pass to validate server.
> > 
> > 
> > The current implementation which i have with the egg is 2 types where in 1, user can have a central directory and then use ptero's mount feature to use the same files for as many servers as you want (i use the mount version mainly and run 4 servers hence saving on disk usage)
> > Mount based egg is for personal use where users manage the server completely
> > Without mount egg will be useful for hosts where they just give the vu installed server to the user and when the user get the server, they will have to fill in their email and password in startup tab, supply gamefiles to `bf3` directory and provide server licence from vu website. Thus making the egg safer to host since Hosts are not gonna supply the files and doesn't affect any TOS of EA.
> 
> Would you provide me with your egg(s)?

#1779

**Comment by Exophobias** [comment]
What I didn’t say is that after this error, the server shuts off and tries rebooting but fails.

**Comment by itsmemac** [comment]
> What I didn’t say is that after this error, the server shuts off and tries rebooting but fails.

Do check the readme. It maybe because you started the server before entering your credentials which in turn created activation file with default credential
[Readme link ](https://github.com/parkervcp/eggs/blob/462a8fba023996b5a67d5bb400723aec77d34329/game_eggs/venice_unleashed/README.md)

**Comment by Exophobias** [comment]
I did not, I put my credentials in when I made the server in the admin panel.

**Comment by itsmemac** [comment]
> I did not, I put my credentials in when I made the server in the admin panel.

Do join [VU discord server](https://discord.gg/dpJwaVZ) and ping me itsmemac#3623
will check this error

**Comment by Exophobias** [comment]
It was an issue with my mount. All good now

**Comment by itsmemac** [comment]
> It was an issue with my mount. All good now

Cool. You should mount the BF3 mount for each servers you run and also assign the mount to the egg.
